### DISCUSSION TITLE
RabbitMQ Cluster Operator unable to add label to node service

### DISCUSSION THREAD

**Discussion by GSS-HLadha** [rabbitmq/cluster-operator#1469] [Q&A] [answered]
Hi,

I am trying to add a label to the rabbitmq cluster operator services, however when i redeploy  after adding a service label the error indicates it is missing from spec:

[- apiVersion: rabbitmq.com/v1beta1
  kind: RabbitmqCluster](error: error validating "RABBITMQCLUSTER.yaml": error validating data: ValidationError(RabbitmqCluster.spec.service): unknown field "labels" in com.rabbitmq.v1beta1.RabbitmqCluster.spec.service; if you choose to ignore these errors, turn validation off with --validate=false)

Note: i am able to provide annotations but not labels to the services

thanks,
Hitesh

**Comment by mkuratczyk** [comment, marked as answer]
As the error says - there's simply no such field. We never supported adding labels on the service level directly.
You an set labels on the RabbitmqCluster level and they'll be propagated to the service:
https://www.rabbitmq.com/kubernetes/operator/using-operator.html#labels-annotations

**Reply by GSS-HLadha** [reply]
Hi Michal,
Thanks for the confirmation.  I am thinking if there is a way via a template i can use the service name to enable the label for the *-node service.  The purpose for this is to have it ignored by consul and only have the one service registered

thanks in advance,
Hitesh
